### PR TITLE
Tests added for remote sensing widget

### LINKED ISSUE #1575: Create proper tests for remote sensing widget
This widget is nearly fully untested causing some issues in the last stable release.
The widget requires interaction with a flight path and the top view, so several components need to be instantiated for testing (or mocked).

Tests should involve flight paths crossing date line and having points of identical coordinates.

### PR DESCRIPTION
**Pull request by rootxrishabh**
fixes #1575

### CODE REVIEW

File: tests/_test_msui/test_remotesensing.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
     This file is part of MSS.
 
     :copyright: Copyright 2017 Reimar Bauer, Joern Ungermann
+    :copyright: Copyright 2023 rootxrishabh
```
Comment (ReimarBauer) [inline tests/_test_msui/test_remotesensing.py]: maybe you want to add your name to the AUTHORS file?

File: tests/_test_msui/test_remotesensing.py [hunk 2]
```diff
@@ -27,17 +28,250 @@
...
 from mslib.msui.remotesensing_dockwidget import RemoteSensingControlWidget
 import skyfield_data
+from PyQt5 import QtWidgets
+from mslib.msui import mpl_qtwidget as qt
+import datetime
+import sys
+from unittest.mock import Mock
+import matplotlib as mpl
+from matplotlib.collections import LineCollection
+app = QtWidgets.QApplication(sys.argv)
```
Comment (ReimarBauer) [inline tests/_test_msui/test_remotesensing.py]: see test results, this won't work
https://github.com/Open-MSS/MSS/actions/runs/4713789246/jobs/8359708435?pr=1739#step:9:30

File: tests/_test_msui/test_remotesensing.py (hunk 1 above)
Comment (ReimarBauer) [inline tests/_test_msui/test_remotesensing.py]: thought adding to authors ...

Comment (rootxrishabh) [reply tests/_test_msui/test_remotesensing.py]: I will remove it after checking the test results.

File: tests/_test_msui/test_remotesensing.py [hunk 3]
```diff
@@ -27,17 +28,250 @@
 
 from mslib.msui.remotesensing_dockwidget import RemoteSensingControlWidget
 import skyfield_data
+from PyQt5 import QtWidgets
+from mslib.msui import mpl_qtwidget as qt
+import datetime
+import sys
```
Comment (ReimarBauer) [inline tests/_test_msui/test_remotesensing.py:30]: there is a pythonic order of imports see PEP8

File: tests/_test_msui/test_remotesensing.py [hunk 4]
```diff
@@ -34,10 +42,238 @@ def test_skyfield_data_expiration(recwarn):
...
+                                                      (79.54503141482452, 28.286788668262982),
+                                                      (79.52243223416616, 28.37325411498613),
+                                                      (79.49980180770042, 28.459715971006098),
+                                                      (79.47713995693299, 28.546174217675254),
+                                                      (79.45444650253313, 28.632628836281985),
+                                                      (79.43172126432714, 28.71907980805177),
+                                                      (79.40896406129221, 28.80552711414579),
+                                                      (79.38617471155024, 28.89197073566029),
+                                                      (79.3633530323609, 28.978410653627137)]
+        self.result_test_tangent_point_coordinates = [(round(x, 2), round(y, 2)) for x, y in
```
Comment (joernu76) [inline tests/_test_msui/test_remotesensing.py]: I'd think it is reasonable to store the data only with two significant digits. This is very cluttered and suggests an unneeded precision.

File: tests/_test_msui/test_remotesensing.py [hunk 5]
```diff
@@ -56,3 +292,61 @@ def test_view_angles(self):
         angle = compute_view_angles(0, 0, 0, 0, -1, 0, 0, -1)
         assert angle[0] == 180.0
         assert angle[1] == -1
+
+    def test_body_angle(self):
+        compute_body_angle = RemoteSensingControlWidget(view=self.view).compute_body_angle
+        angle = compute_body_angle("sun", 73.56, 78.01, 25.27)
+        assert round(angle[0], 2) == 106.71
```
Comment (joernu76) [inline tests/_test_msui/test_remotesensing.py]: Please use pytest approx
https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-approx

Comment (joernu76) [reply tests/_test_msui/test_remotesensing.py]: I'd also suggest to use pytest.mark.parametrize so separate the different tests:
https://docs.pytest.org/en/7.1.x/how-to/parametrize.html#parametrize

File: tests/_test_msui/test_remotesensing.py [hunk 6]
```diff
@@ -34,10 +42,238 @@ def test_skyfield_data_expiration(recwarn):
...
+        self.wp_vertices = [(0, 0), (1, 4)]
+        self.wp_heights = [0, 1000]
+        self.coordinates = [[79.083, 21.15], [77.103, 28.566]]
+        self.heights = [0.0, 0.0]
+        self.times = [datetime.datetime(2023, 4, 15, 10, 9, 59, 174000),
+                      datetime.datetime(2023, 4, 15, 11, 18, 27, 735581)]
+        self.solar_type = ('sun', 'total (horizon)')
 
     def test_view_angles(self):
         compute_view_angles = RemoteSensingControlWidget.compute_view_angles
```
Comment (joernu76) [inline tests/_test_msui/test_remotesensing.py]: The widget is almost always used. Please define a widget as member in setup, or is there a reason not to?

File: tests/_test_msui/test_remotesensing.py [hunk 7]
```diff
@@ -56,3 +292,61 @@ def test_view_angles(self):
...
+        result = result.compute_solar_lines(self.bmap, self.coordinates, self.heights, self.times, self.solar_type)
+        assert isinstance(result, LineCollection)
+
+    def test_tangent_point_coordinates(self):
+        tangent_point_coordinates = RemoteSensingControlWidget(view=self.view).tangent_point_coordinates
+        coordinates = tangent_point_coordinates(lon_lin=self.lon_lin, lat_lin=self.lat_lin, cut_height=self.cut_height)
+        result = [(round(x, 2), round(y, 2)) for x, y in coordinates]
+        assert result == self.result_test_tangent_point_coordinates
+
+    def test_calc_view_rating(self):
```
Comment (joernu76) [inline tests/_test_msui/test_remotesensing.py]: Again, please use "parametrize".

**Review by joernu76 — CHANGES_REQUESTED** [review]

**Comment by ReimarBauer** [comment]
please have a look on the merge conflict and solve it.

**Review by joernu76 — APPROVED** [review]

**Pull request merged**